parsl/dataflow/dflow.py

```diff
@@ -1047,6 +1047,7 @@ def wait_for_current_tasks(self) -> None:
 
         logger.info("Waiting for all remaining tasks to complete")
 
+        # TODO (wardlt): Can this be replaced with conccurent.futures.wait?
```

Comment:
yes I think so

Reply:
There's two pieces of logic here - one is the waiting for app_fu to complete, and one is waiting for that status in the task record to be a final state. Parsl does not cleanly define which of those really means a task is all completed. See issue https://github.com/Parsl/parsl/issues/1279 for related discussion about task completion.

It's important in this `wait_for_current_tasks` call that both of those completion states have been reached for each task.

Reply:
Oh dang, that's interesting about the two termination conditions. I'll retract my TODO comment